### ISSUE TITLE
have receptor --version from binary built from non-tagged release report git hash

### ISSUE THREAD

**Issue by kdelee** [ansible/receptor#498]
When testing binaries of `receptor` built from PRs etc (e.g. not tagged releases), it would be great if `receptor --version` would report a git hash.

Right now it says "Version unknown"

**Comment by eqrx** [comment]
Good idea, we could use 
```
export GITTAG=$(shell git describe --tags --always)
export GITCOMMIT=$(shell git log -1 --pretty=format:"%H")
export GOLDFLAGS=-X [...].version=$(GITTAG) -X [...].commit=$(GITCOMMIT)
```
for that.

**Comment by fosterseth** [comment]
@rooftopcellist did some rearrangement on the downstream build process to get the proper version injections, the version should appear correctly in upcoming builds

**Comment by rooftopcellist** [comment]
tl;dr, the path to the new version file changed and needed to be updated downstream at the point where the project gets compiled.  A new build should be ready in about an hour.

**Comment by rooftopcellist** [comment]
This appears to be fixed now with the new build:

```
$ receptor --version
1.1.1
```

**Comment by kdelee** [comment]
Ok I'm glad it is reporting a version downstream....but last time I used a binary that @fosterseth sent me that was on s3, it also reported the `version Unkown` thing. So it would be great if we got the git hash from upstream builds. So, as much as I'm glad downstream is fixed, I think upstream still is not going to set the git hash until what @eqrx mentioned is done

**Comment by Akasurde** [comment]
@eqrx I get from devel version 

```
./receptor --version
0.9.8.dev294-0.git20220216.3000b61
```

is this the output we are expecting?

**Comment by eqrx** [comment]
Hey @Akasurde! the makefile creates a version string according by doing this:

`$(shell cd receptorctl && python3 setup.py --version)-$(RELEASE)0.git$(shell date +'%Y%m%d').$(shell git rev-parse --short HEAD)`

... no idea why that is done that way but if you wonder why it sees `0.9.8` as the last tag .... for some reason our releases are tagged with lightweight tags. `git rescribe` and friends only consider annotated release tags. lightweight tags should only be used for local operations and not for releasing for multiple reasons, the biggest one it that they only a ref and not a commit (== no message)

**Comment by eqrx** [comment]
Something is going on within the repos commit history, somehow the newer tags are not connected to the devel branch. let me check that out

**Comment by eqrx** [comment]
Regarding the output: @ghjm implemented building the version string like this:

`$(shell cd receptorctl && python3 setup.py --version)-0.git$(shell date +'%Y%m%d').$(shell git rev-parse --short HEAD)`

Not sure why this string includes the current date since it makes reproducible builds impossible. Also not sure why the setuptools version output is used as receptor version

**Comment by eqrx** [comment]
Never have seen this ... the newer tags are annotated ones but I get

``` $ git describe --long
fatal: No annotated tags can describe '745a1e1364a6fed3f90d5cff78ec3dd7a9759c1d'.
However, there were unannotated tags: try --tags.
```

**Comment by eqrx** [comment]
@shanemcd were the new release tags created via the `git` command or via the GH web interface?

**Comment by eqrx** [comment]
FYI: Go v1.18 will include source version information automatically!

**Comment by ghjm** [comment]
> Not sure why this string includes the current date since it makes reproducible builds impossible. Also not sure why the setuptools version output is used as receptor version

This is partly the result of negotiations around a crazy build integration with another team, that long ago stopped being relevant.  By all means, replace this with something more sensible.

**Comment by PabloHiro** [comment]
This is obsolete for the current build system. If a commit has a version tag, `receptor --version` prints:
```
v1.5.7
```
If not tag is present, it uses the latest version and suffixes the shortened commit:
```
v1.5.7+git1c13f32
```
